usaspending_api/search/v2/elasticsearch_helper.py

```diff
@@ -66,8 +71,65 @@ def search_transactions(filters, fields, sort, order, lower_limit, limit):
     index_name = index_name[:-1]+'*'
     try:
         response = CLIENT.search(index=index_name, body=query)
-    except Exception as es1:
+    except Exception:
         return False, -1
     total = response['hits']['total']
     results = format_for_frontend(response['hits']['hits'])
-    return results, total
+    return results, total, transaction_type
+
+
+def get_total_results(keyword, index_name):
+    index_name = '{}-{}'.format(TRANSACTIONS_INDEX_ROOT,
+                                index_name.replace('_', ''))+'*'
+    query = {
+        'query': {
+            'query_string': {
+                'query': keyword
+                }
+            }
+    }
+    try:
+        response = CLIENT.search(index=index_name, body=query)
+        return response['hits']['total']
+    except Exception as es1:
+        return es1
+
+
+def spending_by_transaction_count(filters):
+    keyword = filters['keyword']
+    response = {}
+    for category in award_categories:
+        response[category] = get_total_results(keyword, category)
+    return response
+
+
+def search_keyword_id_list_all(keyword):
+    """
+    althought query size has been increased,
+    scrolling may still be time consuming.
+
+    Timeout has been implemented in the
+    client to prevent the connection
+    from timing out.
+    """
+    index_name = '{}-'.format(TRANSACTIONS_INDEX_ROOT.replace('_', ''))+'*'
+
+    query = {
+        "query": {
+            "query_string": {
+                "query": keyword
+            }
+        }, "size": DOWNLOAD_QUERY_SIZE}
+    try:
+        responses = CLIENT.search(index=index_name, body=query, timeout='3m')
+    except Exception as es1:
+        return es1
```

Comment:
Returning the exception isn't a good practice. Instead could the exception be logged and a None or empty list be returned?

Reply:
Noted, empty list returned and logged.